### PR TITLE
may have found a way to slow the vector animations

### PR DESCRIPTION
**Pull request by AndyLoft**
please test first. there may be better ways but it seems to work for me. the vector animations are tied to the same VBL timer as the lighting.

### CODE REVIEW

**Comment by AndyLoft** [comment]
every now and then the animation loop locks up. any ideas?

**Comment by mheyer32** [comment]
I guess you need to test for "less or equal to zero" as the VBL will decrease the counter asynchronously. It can happen that the counter reaches below 0 before the main animation routine gets around to check it for zero. Now you're essentially waiting for thousands/billions of frames before it can reach 0 again...

**Comment by 0xABADCAFE** [comment]
> I guess you need to test for "less or equal to zero" as the VBL will decrease the counter asynchronously. It can happen that the counter reaches below 0 before the main animation routine gets around to check it for zero. Now you're essentially waiting for thousands/billions of frames before it can reach 0 again...

A bit like waiting for a train here in the UK just now...

**Comment by AndyLoft** [comment]
last commit seems promising but requires more testing.

File: ab3d2_source/newaliencontrol.s [hunk 1]
```diff
@@ -918,9 +913,12 @@ ACTANIMOBJ:
 				moveq	#0,d1
 				move.b	5(a3,d0.w),d1
 
-				cmp	#0,vecttimer		;animtimer decriment moved to VBlankInterrupt:
+				cmp	#0,vecttimer
+				ble.s	.zero
+				cmp	#0,vecttimer
 				bgt.s	.notzero
-
+.zero
```
Comment (mheyer32) [inline ab3d2_source/newaliencontrol.s:920]: whats wrong with just bgt.s ?

**Comment by 0xABADCAFE** [comment]
So, I'm going to be that guy. I think that we reset values were putting into vecttimer and animmtimer should themselves be global values that we can modify rather than the immediate values in the code (assuming I'm reading it correctly).

Also, when defining new data symbols, can we use the domain_name_type convention? For example, Anim_VecTimer_w - tells us immediately this belongs to animation, is "public" (i.e. is referenced from multiple locations) and is a word type/size.

**Comment by AndyLoft** [comment]
I get confused by some of the terminology and so will have to improve.

File: ab3d2_source/newaliencontrol.s [hunk 2]
```diff
@@ -913,9 +913,9 @@ ACTANIMOBJ:
 				moveq	#0,d1
 				move.b	5(a3,d0.w),d1
 
-				cmp	#0,animtimer		;animtimer decriment moved to VBlankInterrupt:
+				cmp	#0,vecttimer		;animtimer decriment moved to VBlankInterrupt:
```
[the remark this answers is not in the document]
Comment (AndyLoft) [reply ab3d2_source/newaliencontrol.s]: done now. I think

File: ab3d2_source/newaliencontrol.s [hunk 3]
```diff
@@ -893,8 +893,13 @@ ACTANIMOBJ:
 
 				moveq	#0,d1
 				move.b	5(a3,d0.w),d1
-				move.w	d1,EntT_Timer1_w(a0)
 
+				cmp	#0,vecttimer
+				bgt.s	.nozero
+
+				move.w	#3,vecttimer
```
[the remark this answers is not in the document]
Comment (AndyLoft) [reply ab3d2_source/newaliencontrol.s]: this was a mistake, have now skipped 'move.b 5(a3,d0.w),d1' but it has not helped nor hindered the issue I am seeing.
will keep digging

File: ab3d2_source/newaliencontrol.s (hunk 1 above)
Comment (AndyLoft) [reply ab3d2_source/newaliencontrol.s]: nothing, the extra check for less than was a 'just in case'.

**Comment by AndyLoft** [comment]
I think I may have put the set/reset in the wrong place, should maybe go after the fps limiter?

**Comment by AndyLoft** [comment]
this is not the way forward.
